extension/src/main/java/io/camunda/zeebe/process/test/extension/ZeebeProcessTestExtension.java

```diff
@@ -18,13 +18,24 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
+ * This extension is used by the {@link ZeebeProcessTest} annotation. It is responsible for managing
+ * the lifecycle of the test engine.
+ */
 public class ZeebeProcessTestExtension
     implements BeforeEachCallback, AfterEachCallback, TestWatcher {
 
   private static final Logger LOG = LoggerFactory.getLogger(ZeebeProcessTestExtension.class);
   private static final String KEY_ZEEBE_CLIENT = "ZEEBE_CLIENT";
   private static final String KEY_ZEEBE_ENGINE = "ZEEBE_ENGINE";
 
+  /**
+   * Before each test a new test engine gets created and started. A client to communicate with the
+   * engine will also be created. Together with a {@link RecordStream} these will be injected in the
+   * fields of the test class, if they are available.
+   *
+   * @param extensionContext jUnit5 extension context
+   */
```

Comment:
🔧 Maybe link this with an `@see` or so to documentation about the injected fields.